src/Engines/BaseEngine.cs

```diff
@@ -202,31 +200,39 @@ public void RegisterDefaultEncoders()
 
         internal MimeBundle EncodeForDisplay(object displayable)
         {
-            if (displayable == null) throw new ArgumentNullException(nameof(displayable));
             // Each serializer contributes what it can for a given object,
             // and we take the union of their contributions, with preference
             // given to the last serializers registered.
             var displayData = MimeBundle.Empty();
+
             foreach ((var mimeType, var encoders) in serializers)
             {
                 foreach (var encoder in encoders)
                 {
-                    var encoded = encoder.Encode(displayable);
-                    if (encoded == null)
-                    {
-                        continue;
-                    }
-                    else
+                    try
                     {
-                        displayData.Data[mimeType] = encoded.Value.Data;
-                        if (encoded.Value.Metadata != null)
+                        var encoded = encoder.Encode(displayable);
+                        if (encoded == null)
+                        {
+                            continue;
+                        }
+                        else
                         {
-                            displayData.Metadata[mimeType] = encoded.Value.Metadata;
+                            displayData.Data[mimeType] = encoded.Value.Data;
+                            if (encoded.Value.Metadata != null)
+                            {
+                                displayData.Metadata[mimeType] = encoded.Value.Metadata;
+                            }
+                            break;
                         }
-                        break;
+                    }
+                    catch (Exception e)
+                    {
+                        this.Logger.LogWarning($"Encoder {encoder.GetType().FullName} threw an exception encoding '{displayable}' ({e.Message}).");
```

Comment:
Do we want to log the exception as well in this case?
```suggestion
                        this.Logger.LogWarning(e, $"Encoder {encoder.GetType().FullName} threw an exception encoding '{displayable}' ({e.Message}).");
```